### PR TITLE
added AdversarialBiasMitigator tests and model

### PR DESCRIPTION
**Pull request by ArjunSubramonian**
- Added `AdversarialBiasMitigator` tests.
- Added `adversarial-binary-gender-bias-mitigated-roberta-snli` model.

Depends on: https://github.com/allenai/allennlp/pull/5269.

### CODE REVIEW

**Comment by dirkgr** [comment]
I'm not reviewing this until allenai/allennlp#5269 is merged, right?

File: allennlp_models/modelcards/pair-classification-adversarial-binary-gender-bias-mitigated-roberta-snli.json [hunk 1]
```diff
@@ -0,0 +1,70 @@
+{
+    "id": "pair-classification-adversarial-binary-gender-bias-mitigated-roberta-snli",
+    "registered_model_name": "adversarial_bias_mitigator",
+    "registered_predictor_name": "textual_entailment",
+    "display_name": "Adversarial Binary Gender Bias-Mitigated RoBERTa SNLI",
+    "task_id": "textual_entailment",
+    "model_details": {
+        "description": "This `Model` implements a basic text classifier and feedforward regression adversary with an adversarial bias mitigator wrapper. The text is embedded into a text field using a RoBERTa-large model. The resulting sequence is pooled using a cls_pooler `Seq2VecEncoder` and then passed to a linear classification layer, which projects into the label space. Subsequently, a `FeedForwardRegressionAdversary` attempts to recover the coefficient of the static text embedding in the binary gender bias subspace. While the adversary's gradients are computed normally, the classifier's gradients are computed such that updates to the classifier's parameters will not aid the adversary and will make it more difficult for the adversary to recover binary gender bias.",
```
Comment (dirkgr) [inline allennlp_models/modelcards/pair-classification-adversarial-binary-gender-bias-mitigated-roberta-snli.json:8]: If this method really doesn't work that well, as you said in standup, then I think we should have a description of that somewhere.

Comment (ArjunSubramonian) [reply allennlp_models/modelcards/pair-classification-adversarial-binary-gender-bias-mitigated-roberta-snli.json]: The method actually ended up working quite well.

File: allennlp_models/modelcards/pair-classification-adversarial-binary-gender-bias-mitigated-roberta-snli.json [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        "dataset": {
+            "name": "Stanford Natural Language Inference (SNLI) train set",
+            "url": "https://nlp.stanford.edu/projects/snli/",
+            "processed_url": "https://allennlp.s3.amazonaws.com/datasets/snli/snli_1.0_train.jsonl"
+        },
+        "motivation": null,
+        "preprocessing": null
+    },
+    "quantitative_analyses": {
+        "unitary_results": null,
```
Comment (AkshitaB) [inline allennlp_models/modelcards/pair-classification-adversarial-binary-gender-bias-mitigated-roberta-snli.json:56]: I'm curious; what numbers did we get?

**Review by AkshitaB — APPROVED** [review]

File: allennlp_models/modelcards/pair-classification-adversarial-binary-gender-bias-mitigated-roberta-snli.json (hunk 2 above)
Comment (ArjunSubramonian) [reply allennlp_models/modelcards/pair-classification-adversarial-binary-gender-bias-mitigated-roberta-snli.json]: Added!

**Pull request merged**